supercompress/client.py

```
@@ -165,4 +165,136 @@ def __exit__(self, *args: object) -> None:
+                data["keep_ratio"]
+                if data.get("keep_ratio") is not None
+                else budget_ratio if budget_ratio is not None
+                else 0.35
+            ),
+            cache_prefix_applied=bool(data.get("cache_prefix_applied", False)),
+            compression_risk=data.get("compression_risk"),
+            confidence=data.get("confidence"),
+            ccr=data.get("ccr"),
+        )
+
+    async def retrieve(self, hash: str) -> Optional[str]:
```

Ruff (0.16.1)

[error] Function argument `hash` is shadowing a Python builtin

(A002)